src/binding.cc

```diff
@@ -441,6 +428,45 @@ void WriteInt64(const CallbackInfo& args) {
   *reinterpret_cast<int64_t*>(ptr) = val;
 }
 
+
+void WriteInt32(const CallbackInfo& args) {
+  Env env = args.Env();
+  char* ptr = AddressForArgs(args);
+
+  Value in = args[2];
+  int64_t val;
+  if (in.IsNumber()) {
+    val = in.As<Number>();
+  } else if (in.IsString()) {
+    char* endptr;
+    char* str;
+    int base = 0;
+    std::string _str = in.As<String>();
+    str = &_str[0];
+
+    errno = 0;     /* To distinguish success/failure after call */
+    val = strtoll(str, &endptr, base);
+
+    if (endptr == str) {
+      throw TypeError::New(env, "writeInt32: no digits we found in input String");
+    } else if (errno == ERANGE && (val == INT32_MAX || val == INT32_MIN)) {
+      throw TypeError::New(env, "writeInt32: input String numerical value out of range");
+    } else if (errno != 0 && val == 0) {
+      char errmsg[200];
+      snprintf(errmsg, sizeof(errmsg), "writeInt32: %s", strerror(errno));
+      throw TypeError::New(env, errmsg);
+    }
+  } else {
+    throw TypeError::New(env, "writeInt32: Number/String 32-bit value required");
+  }
+
+  if (val < INT32_MIN || val > INT32_MAX) {
+      throw TypeError::New(env, "writeInt32: value out of range");
+  }
+
+  *reinterpret_cast<int32_t*>(ptr) = (int32_t)val;
```

Comment:
Ditto here – but I would also really maybe reconsider implementing this in C++, it would be nice to have all the string parsing etc. done in JS and then just write 4 bytes into a temporary `Buffer` here + provide a method to write a Buffer into a `PointerBuffer` at a given offset